src/log.cpp

```diff
@@ -187,13 +190,48 @@ void set_log_to_file()
 	if(is_log_dir_writable_.value_or(false)) {
 		// get the log file stream and assign cerr+cout to it
 		output_file_path_ = filesystem::get_logs_dir()+"/"+unique_log_filename();
-		static std::unique_ptr<std::ostream> logfile { filesystem::ostream_file(output_file_path_) };
-		static std::ostream cerr_stream{std::cerr.rdbuf()};
-		//static std::ostream cout_stream{std::cout.rdbuf()};
-		auto cerr_tee { boost::iostreams::tee(*logfile, cerr_stream) };
-		output_file_.reset(new boost::iostreams::stream<decltype(cerr_tee)>{cerr_tee, 4096, 0});
-		std::cerr.rdbuf(output_file_.get()->rdbuf());
-		std::cout.rdbuf(output_file_.get()->rdbuf());
+
+		// redirect to a file
+		// IMPORTANT: apparently redirecting stderr/stdout will also redirect std::cerr/std::cout, but the reverse is not true
+		//            redirecting std::cerr/std::cout will *not* redirect stderr/stdout
+		if(setvbuf(stdout, NULL, _IONBF, 2) == -1) {
+			std::cerr << "Failed to set stdout to be unbuffered";
+		}
+
+// because apparently the number of leading underscores is of such critical importance...
+#ifdef _WIN32
+		int fd = _open(output_file_path_.c_str(), O_RDWR | O_CREAT, _S_IREAD | _S_IWRITE);
+		// Windows doesn't define these anywhere
+		#define	STDOUT_FILENO 1
+		#define	STDERR_FILENO 2
+#elif defined __APPLE__
+		int fd = open(output_file_path_.c_str(), O_RDWR | O_CREAT, S_IREAD | S_IWRITE);
+#else
+		int fd = open(output_file_path_.c_str(), O_RDWR | O_CREAT, __S_IREAD | __S_IWRITE);
+#endif
+		if(fd != -1) {
+			if(dup2(fd, STDOUT_FILENO) == -1) {
+				std::cerr << "Failed to redirect stdout";
+			}
+			if(dup2(fd, STDERR_FILENO) == -1) {
+				std::cerr << "Failed to redirect stderr";
+			}
+		} else {
+			std::cerr << "Failed to create log file: " << output_file_path_;
+		}
+
+		// std::fflush(stderr);
```

Comment:
Comments should be deleted.

Reply:
That's the plan - it's there for information. Also I forgot to set this as a draft.